File: csharp/Svix/Models/EndpointOut.cs
// this file is @generated
using System.Text;
using Newtonsoft.Json;

namespace Svix.Models
{
    public class EndpointOut
    {
        [JsonProperty("channels")]
        public List<string>? Channels { get; set; } = null;

        [JsonProperty("createdAt", Required = Required.Always)]
        public required DateTime CreatedAt { get; set; }

        [JsonProperty("description", Required = Required.Always)]
        public required string Description { get; set; }

        [JsonProperty("disabled")]
        public bool? Disabled { get; set; } = null;

        [JsonProperty("filterTypes")]
        public List<string>? FilterTypes { get; set; } = null;

        [JsonProperty("id", Required = Required.Always)]
        public required string Id { get; set; }

        [JsonProperty("metadata", Required = Required.Always)]
        public required Dictionary<string, string> Metadata { get; set; }

        [JsonProperty("rateLimit")]
        public ushort? RateLimit { get; set; } = null;

        [JsonProperty("uid")]
        public string? Uid { get; set; } = null;

        [JsonProperty("updatedAt", Required = Required.Always)]
        public required DateTime UpdatedAt { get; set; }

        [JsonProperty("url", Required = Required.Always)]
        public required string Url { get; set; }

        [JsonProperty("version", Required = Required.Always)]
        public required int Version { get; set; }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();

            sb.Append("class EndpointOut {\n");
            sb.Append("  Channels: ").Append(Channels).Append('\n');
            sb.Append("  CreatedAt: ").Append(CreatedAt).Append('\n');
            sb.Append("  Description: ").Append(Description).Append('\n');
            sb.Append("  Disabled: ").Append(Disabled).Append('\n');
            sb.Append("  FilterTypes: ").Append(FilterTypes).Append('\n');
            sb.Append("  Id: ").Append(Id).Append('\n');
            sb.Append("  Metadata: ").Append(Metadata).Append('\n');
            sb.Append("  RateLimit: ").Append(RateLimit).Append('\n');
            sb.Append("  Uid: ").Append(Uid).Append('\n');
            sb.Append("  UpdatedAt: ").Append(UpdatedAt).Append('\n');
            sb.Append("  Url: ").Append(Url).Append('\n');
            sb.Append("  Version: ").Append(Version).Append('\n');
            sb.Append("}\n");
            return sb.ToString();
        }
    }
}

File: ruby/lib/svix/models/endpoint_out.rb
# frozen_string_literal: true
# This file is @generated
require "json"

module Svix
  class EndpointOut
    attr_accessor :channels
    attr_accessor :created_at
    attr_accessor :description
    attr_accessor :disabled
    attr_accessor :filter_types
    attr_accessor :id
    attr_accessor :metadata
    attr_accessor :rate_limit
    attr_accessor :uid
    attr_accessor :updated_at
    attr_accessor :url
    attr_accessor :version

    def initialize(attributes = {})
      unless attributes.is_a?(Hash)
        fail ArgumentError, "The input argument (attributes) must be a hash in `Svix::EndpointPatch` new method"
      end
      attributes.each do |k, v|
        instance_variable_set "@#{k}", v
      end
    end

    def self.deserialize(attributes = {})
      attributes = attributes.transform_keys(&:to_s)
      attrs = {
        'channels': attributes["channels"],
        'created_at': attributes["createdAt"],
        'description': attributes["description"],
        'disabled': attributes["disabled"],
        'filter_types': attributes["filterTypes"],
        'id': attributes["id"],
        'metadata': attributes["metadata"],
        'rate_limit': attributes["rateLimit"],
        'uid': attributes["uid"],
        'updated_at': attributes["updatedAt"],
        'url': attributes["url"],
        'version': attributes["version"],
      }
      new attrs
    end

    def serialize
      out = Hash.new
      out["channels"] = @channels
      out["createdAt"] = @created_at
      out["description"] = @description
      out["disabled"] = @disabled
      out["filterTypes"] = @filter_types
      out["id"] = @id
      out["metadata"] = @metadata
      out["rateLimit"] = @rate_limit
      out["uid"] = @uid
      out["updatedAt"] = @updated_at
      out["url"] = @url
      out["version"] = @version
      out.compact
    end

    # Serializes the object to a json string
    # @return String
    def to_json
      JSON.dump(serialize)
    end
  end
end
